### ISSUE TITLE
Support `string.coerceTo('array')` and `array.coerceTo('string')`

### ISSUE THREAD

**Issue by coffeemug** [rethinkdb/rethinkdb#1147]
Currently there are many commands missing on strings that are present on arrays. Until we get around to filling out the string API, we should allow coercing to arrays and back as an option to do things like `slice`:

``` javascript
r.expr('foo').coerceTo('array') // ['f', 'o', 'o']
r.expr(['f', 'o', 'o']).coerceTo('string') // 'foo'
```

**Comment by srh** [comment]
str.coerceTo('array') can be done with str.split('').  Closing.

**Comment by srh** [comment]
You could do `r.add(r.args(['f', 'o', 'o']))`.  I think there should be a string join function though.  The question is, what should it be named in a database engine?

**Comment by thelinuxlich** [comment]
Maybe r.concat?

**Comment by srh** [comment]
Apparently I was wrong about `split` because it does crazy stuff with combining characters, such as combining them.  Reopening.

**Comment by srh** [comment]
Maybe we could call it `unsplit`, because `concat` sounds related to `concatMap`, or vice versa.  `r.expr(":").unsplit(["foo", "bar"])` => `"foo:bar"`.  (And `join` is obviously a bad idea.)

**Comment by thelinuxlich** [comment]
Unsplit is good

**Comment by marshall007** [comment]
Yea, I like `unsplit` too, but I think the string separator argument should be second and optional. I would have said the separator argument should default to `,` but since `.split()` defaults to "splitting on whitespace" I'm not sure what we should do.

**Comment by srh** [comment]
Maybe the separator should be second, but not optional.

**Comment by srh** [comment]
My serious inclination is to have the separator be second, default value is the empty string.  So `.unsplit()` just concatenates strings.

**Comment by marshall007** [comment]
@srh yep, sounds good. That's how I wish `.split()` behaved, but by default it splits on all whitespace. It would seem reasonable that `<string>.split().unsplit() == <string>`, but unfortunately that will not always be the case.

**Comment by gabor-boros** [comment]
@srh Reading through the thread about, the title could be "ReQL proposal: `unsplit` command" since it is more about implementing that rather than extending coercions, right?
